test/MathUtils.test.cpp

```diff
@@ -218,3 +218,31 @@ TEST(MathUtils, isRectInRectClass)
 	EXPECT_EQ(false, NAS2D::isRectInRect({5, 5, 1, 1}, {3, 3, 1, 1}));
 	EXPECT_EQ(false, NAS2D::isRectInRect({5, 5, 1, 1}, {7, 3, 1, 1}));
 }
+
+TEST(MathUtils, mapDomainToRange)
+{
+	//I know the temperature conversions have established formulas, but that's not what I'm testing.
+	//Given proper input I should get the same value as if I used the derived formulas.
+
+	//(v - minD) * (maxR - minR) / (maxD - minD) + minR;
+	//(F - 32.0f) * (5.0f / 9.0f) + 0.0f;
+	//(C - 0.0f) * (9.0f / 5.0f) + 32.0f;
+
+	//Fahrenheit to Celcius
+	EXPECT_NEAR(0.0f, NAS2D::convertDomainToRange(32.0f, 32.0f, 41.0f, 0.0f, 5.0f), 0.0001f);
+	EXPECT_NEAR(100.0f, NAS2D::convertDomainToRange(212.0f, 32.0f, 41.0f, 0.0f, 5.0f), 0.0001f);
+
+	//Celcius to Fahrenheit
+	EXPECT_NEAR(32.0f, NAS2D::convertDomainToRange(0.0f, 0.0f, 5.0f, 32.0f, 41.0f), 0.0001f);
+	EXPECT_NEAR(212.0f, NAS2D::convertDomainToRange(100.0f, 0.0f, 5.0f, 32.0f, 41.0f), 0.0001f);
+
+	//char to normalized float
+	EXPECT_NEAR(0.0f, NAS2D::convertDomainToRange(0.0f, 0.0f, 255.0f, 0.0f, 1.0f), 0.01f);
```

Comment:
To be overly pedantic, the comment should read `unsigned char`, since we are assuming a range of `0..255`. The standard makes no guarantee if `char` is `signed` or `unsigned`.

If the template is updated with two type parameters, the inputs could be given as actual `unsigned char` or `int`, rather than as `float`.

----

More generally, for all `EXPECT` calls, usually the expected value is given first on the left, while the method being tested is on the right. Most other unit tests in the project are written in that order. Here the order is swapped.

Reply:
Noted on the `unsigned char`.

The expected value *is* on the left and the method is on the right.

`Expect_Near` takes a third parameter: `abs_error` which is the acceptable `epsilon` of the `expected result` vs the `actual result`. Because I knew I was dealing with `float`s, testing exact equality is a logic error.

Reply:
Oh, geeze, I read that wrong. I'm so used to the non-floating point versions of `EXPECT`.